compiler/main.nim

```diff
@@ -54,6 +54,7 @@ proc commandDoc2 =
   finishDoc2Pass(gProjectName)
 
 proc commandCompileToC =
+  extccomp.initVars()
```

Comment:
In https://github.com/Araq/Nim/commit/53f4c7758b0c9d3df562795eeca0a553eb22e219 you removed the `setCC("gcc")`. Now the `.options.always` are read only once with the first call to `initVars`, but then `cpp` is not defined yet, so `gcc.options.always` are used instead of `gcc.cpp.options.always`. Since we don't want to use `setCC("gcc")` so as not to introduce #1601 again, I call `initVars` again, after `cpp` should be set. Maybe there's a better way to do this.